src/fabric.c

```diff
@@ -1207,8 +1215,12 @@ static void ofi_set_prov_attr(struct fi_fabric_attr *attr,
 
 	core_name = attr->prov_name;
 	if (core_name) {
-		assert(ofi_is_util_prov(prov));
-		attr->prov_name = ofi_strdup_append(core_name, prov->name);
+		if (ofi_is_util_prov(prov))
+			attr->prov_name = ofi_strdup_append(core_name, prov->name);
+		else if (ofi_is_lnx_prov(prov))
+			attr->prov_name = ofi_strdup_link_append(core_name, prov->name);
+		else
+			assert(0);
```

Comment:
*** this is going to break on non-debug builds. if the assert is the only thing in the if, it needs brackets so it doesn't pick up the next line on non-debug builds

Reply:
fixed.